explore_utils.py

```diff
@@ -56,3 +56,25 @@ def find_most_common_traffic_sources(dataset,num=5):
 
     """
     return dataset.train['trafficSource.source'].value_counts().head(num)
+
+def social_active_customers_revenue(dataset):
+
+    """Find the revenue generated by socially engaged users, determining
+    whether or not they generate more revenue
+
+    args:
+    dataset (Dataset): the google analytics dataset
+
+    returns:
+    the revenue generated by social engaged users, and compares it to
+    normal users
+
+    """
+    train_df = dataset.train.copy()
+
+    #   Separates data frame into socialEngagementTypes
+    sociallyengaged = train_df.groupby("fullVisitorId")['socialEngagementType'].value()
```

Comment:
`.value()` doesn't exist (this is your bug), but it looks like you're trying to select the users who are socially engaged. You have to be a little more specific for this to work - since we have multiple visits per customer, it's hard to define whether that customer is socially engaged, because they may have some visits where they are engaged and some where they are not. If you want to know if the customer is socially engaged in any of their visits, I think you can do something like this:

```
train_df.groupby("fullVisitorId")['socialEngagementType'].equals('Socially Engaged').any()
```

What this does, left to right:
1. group visitors by id
2. select the socialEngagementType column
3. check if each row contains "Socially Engaged" (I don't know if this is actually the correct string you need to match, please check)
4. checks if any of the rows contained "Socially Engaged"

The result will be a series that matches visitor IDs to a boolean.

Reply:
FYI I didn't test this line so it probably won't work if you paste it in. Please check my work before submitting